packages/core/src/common/k8s-api/__tests__/api-manager.test.ts

```diff
@@ -72,21 +75,48 @@ describe("ApiManager", () => {
         logger: di.inject(loggerInjectable),
       }, kubeApi);
 
-      apiManager.registerApi(apiBase, kubeApi);
+      apiManager.registerApi(kubeApi);
 
       // Define to use test api for ingress store
       Object.defineProperty(kubeStore, "api", { value: kubeApi });
-      apiManager.registerStore(kubeStore, [kubeApi]);
+      apiManager.registerStore(kubeStore);
 
       // Test that store is returned with original apiBase
       expect(apiManager.getStore(kubeApi)).toBe(kubeStore);
 
       // Change apiBase similar as checkPreferredVersion does
       Object.defineProperty(kubeApi, "apiBase", { value: fallbackApiBase });
-      apiManager.registerApi(fallbackApiBase, kubeApi);
+      apiManager.registerApi(kubeApi);
 
       // Test that store is returned with new apiBase
       expect(apiManager.getStore(kubeApi)).toBe(kubeStore);
     });
   });
+
+  describe("technical tests for autorun", () => {
```

Comment:
could add another test where the second api is removed and then `apiManager.getApi(apiBase)` returns `firstApi`?

Reply:
Actually this is not possible given that the apis are registered automatically and then never garbage collected.